src/transactions/ExtendFootprintTTLOpFrame.cpp

```diff
@@ -49,6 +49,139 @@ ExtendFootprintTTLOpFrame::isOpSupported(LedgerHeader const& header) const
     return header.ledgerVersion >= 20;
 }
 
+bool
+ExtendFootprintTTLOpFrame::doPreloadEntriesForParallelApply(
+    Config const& config, SorobanMetrics& sorobanMetrics,
+    AbstractLedgerTxn& ltx, ThreadEntryMap& entryMap, OperationResult& res,
+    SorobanTxData& sorobanData) const
+{
+    ExtendFootprintTTLMetrics metrics(sorobanMetrics);
+
+    auto const& resources = mParentTx.sorobanResources();
+
+    auto callback = [&metrics, &sorobanData, &resources, &res,
+                     &config](LedgerKey const& lk, uint32_t entrySize) -> bool {
+        metrics.mLedgerReadByte += entrySize;
+        if (resources.readBytes < metrics.mLedgerReadByte)
```

Comment:
This flow needs to be updated for extend and restore. We shouldn't load entries in certain cases (like if the entry has been archived here).